### PR TITLE
[Wav2Vec2] SpecAugment Fast

### LINKED ISSUE #10459: [Wav2Vec2] Improve SpecAugment function by converting numpy based function to pytorch based function
# 🚀 Feature request

As can be seen here: https://github.com/huggingface/transformers/blob/11655fafdd42eb56ad94e09ecd84d4dc2d1041ae/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L47,

the function `_compute_mask_indices` (responsible for spec augment) of Wav2Vec2 is written in numpy which means that the function is not GPU compatible. 
The function could simply be rewritten in PyTorch, which should make training on GPU faster.

This "Good First Issue" is about converting `_compute_mask_indices` to PyTorch while keeping the same functionality. 

## Your contribution

I'm happy to guide the contributor through the PR!

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR refactores the SpecAugement implementation for Wav2Vec2 by fully relying on PyTorch instead of numpy. 
1) The code is made more readable 
- `attention_mask` is dropped since it's not required to treat masked batch indices differently
- Previously every batch_idx was forced to have the same number of masked indices (overlapping masked indices can lead to some batch_indices to have fewer masked indices). This is also not enforced here since it would make the function very dependend on the batch_size which is not good IMO. I don't see a reason why different batch_idx cannot have different # of masked indices. It was verified via training that the change does not lead to a performance drop.
2) Replacing a for loop with tensorized code lead to a 1% speed-up in training (not really noticeable thb)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Noticed a small speed-up when training (1-2%) only though, and even slighly improved results. More importantly I think the code is much more readable now.
